app/server/api/poli_payments_api.py

```diff
@@ -0,0 +1,87 @@
+from flask import Blueprint, request, make_response, jsonify
+from flask.views import MethodView
+from server import db
+from server.utils.poli_payments import PoliPaymentsError, get_poli_link_status, get_poli_link_from_token
+from server.models.fiat_ramp import FiatRamp, FiatRampStatusEnum
+from server.models.user import User
+from server.utils.credit_transfers import make_deposit_transfer, find_user_with_transfer_account_from_identifiers
+
+poli_payments_blueprint = Blueprint('poli_payments_blueprint', __name__)
+
+
+class PoliPaymentWebhookAPI(MethodView):
+    def post(self):
+        """
+        POLi Payments Webhook that provides a token to lookup a POLi link and check status.
+
+        :return:
+        """
+
+        if request.data:
+            try:
+                response = get_poli_link_from_token(token=request.data.decode())
+
+            except PoliPaymentsError as e:
+                response_object = {
+                    'message': str(e)
+                }
+                return make_response(jsonify(response_object)), 400
+
+            poli_link = response['poli_link']
+            try:
+                get_poli_link_status_response = get_poli_link_status(
+                    poli_link=poli_link
+                )
+
+            except PoliPaymentsError as e:
+                response_object = {
+                    'message': str(e)
+                }
+                return make_response(jsonify(response_object)), 400
+
+            fiat_ramp = db.session.query(FiatRamp).filter(
+                FiatRamp.payment_metadata["poli_link"].astext == 'https://poli.to/'+poli_link
```

Comment:
this seems a bit fragile. if poli ever changes their url in any way this would break. also it's a bit confusing that fiat ramp's metadata's "poli_link" is not the same as "poli_link" from the response

Reply:
Good point. Will review.

I'll change the variable names at least.